### ISSUE TITLE
SROAP04005: Could not find schema class in index: java.lang.Object with smallrye-openapi &  resteasy-reactive Resource returning RestResponse<?>

### ISSUE THREAD

**Issue by spieps** [quarkusio/quarkus#31744]
### Describe the bug

In `rest-client-reactive-quickstart` if we go to the ExtensionResource and modify the @Blocking GET request to look like this

```
@GET
@Path("/id/{id}")
@Blocking
public RestResponse<?> id(String id) {
    try {
        return RestResponse.status(RestResponse.Status.OK, extensionsService.getById(id));
     } catch (Exception ex) {
        return RestResponse.status(RestResponse.Status.INTERNAL_SERVER_ERROR, ex);
    }
}
```

And we add smallrye-openapi to the pom.xml
```
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-smallrye-openapi</artifactId>
</dependency>
```

Then on application start up with `./mvnw compile quarkus:dev` the following warning is logged

`WARN  [io.sma.ope.run.scanner] (build-33) SROAP04005: Could not find schema class in index: java.lang.Object`

### Expected behavior

No warning is logged

### Actual behavior

Logged warning `SROAP04005: Could not find schema class in index: java.lang.Object`

### How to Reproduce?

1. `rest-client-reactive-quickstart` 
2. Navigate to ExtensionResource and modify the @Blocking GET request to look like this

```
@GET
@Path("/id/{id}")
@Blocking
public RestResponse<?> id(String id) {
    try {
        return RestResponse.status(RestResponse.Status.OK, extensionsService.getById(id));
     } catch (Exception ex) {
        return RestResponse.status(RestResponse.Status.INTERNAL_SERVER_ERROR, ex);
    }
}
```

3. And we add smallrye-openapi to the pom.xml
```
<dependency>
  <groupId>io.quarkus</groupId>
  <artifactId>quarkus-smallrye-openapi</artifactId>
</dependency>
```

### Output of `java -version`

OpenJDK 64-Bit Server VM GraalVM CE 22.3.0 (build 17.0.5+8-jvmci-22.3-b08, mixed mode, sharing)

### Quarkus version or git rev

2.16.4.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0)
Maven home: /usr/local/Cellar/maven/3.8.5/libexec
Java version: 17.0.5, vendor: GraalVM Community, runtime: /Library/Java/JavaVirtualMachines/graalvm-ce-java17-22.3.0/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "13.2.1", arch: "x86_64", family: "mac"

**Comment by spieps** [comment]
Im trying to add the label area/openapi but not able to. 

See similar issue [#20152](https://github.com/quarkusio/quarkus/issues/20152)

**Comment by geoand** [comment]
@phillip-kruger @MikeEdgar do we have support for `RestResponse` in smallrye-openapi?

If not, we should probably add it. `RestResponse` is essentially like `Response` but with a generic argument

**Comment by MikeEdgar** [comment]
@geoand , `RestResponse` is supported, but the wildcard is being handled poorly in this case. I'll take a look. It should be a simple fix.

**Comment by geoand** [comment]
Gotcha, thanks

**Comment by MikeEdgar** [comment]
The fix for this will handle the wildcard more gracefully. Note that the resulting schema will effectively be an "anything goes" schema like `schema: {}` since there isn't any type information to scan. 

You will probably want to provide custom `@APIResponse` annotations on the method to customize the possible response schemas.

**Comment by spieps** [comment]
Is it ok we close this now, or should we wait until the fix is fully released?

This is fixed by #32095 

@gsmet should we add the 3.0 main milestone on this issue?

**Comment by geoand** [comment]
I added it
